spec/requests/host_aggregates_spec.rb

```diff
@@ -67,6 +68,61 @@
     end
   end
 
+  describe 'POST /api/host_aggregates/:id' do
+
+    before do
+      @zone = FactoryBot.create(:zone, :name => "api_zone")
+      @microstack = FactoryBot.create(:ems_cloud)
+      @provider = FactoryBot.create(:provider, :zone => @zone)
+      @microstack.update(:provider => @provider) ## microstack.provider = provider 
+      @infra = FactoryBot.create(:ems_infra, :name => "microstack ironic", :hostname => "ironic", :zone => @zone, :provider => @provider)
+      @cluster = FactoryBot.create(:ems_cluster, :name => "Microstack Compute", :ext_management_system => @infra)
+      @host = FactoryBot.create(:host_with_authentication, :ext_management_system => @infra, :ems_cluster => @cluster, :name => "host", :hostname => "host.ironic.microstack", :vmm_vendor => "openstack_infra")
+    end
+
+    it 'adds a host to the host aggregate' do
+      host_aggregate = FactoryBot.create(:host_aggregate)
+      api_basic_authorize action_identifier(:host_aggregates, :edit)
+
+      stub_supports(host_aggregate.class, :add_host)
+      stub_supports(host_aggregate.class, :add_host_queue)
+
+      @microstack.update(:host_aggregates => [host_aggregate])
+
+      post(api_host_aggregate_url(nil, host_aggregate), :params => {:action => 'add_host', :resource => {:host_id => @host.id.to_s}})
+
+      expected = {
+          'success' => true,
+          'message' => a_string_including('Adding Host Aggregate'),
+          'task_id' => a_kind_of(String)
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+    end
+
+    it 'removes a host from the host aggregate' do
+      host_aggregate = FactoryBot.create(:host_aggregate)
```

Comment:
You can specify the hosts when building the host_aggregate:
```suggestion
      host_aggregate = FactoryBot.create(:host_aggregate, :hosts => [host])
```